webapi/src/main/scala/org/knora/webapi/messages/admin/responder/usersmessages/UsersMessagesADM.scala

```diff
@@ -697,39 +698,55 @@ object UserIdentifierType extends Enumeration {
     val USERNAME = Value(3, "username")
 }
 
-
 /**
-  * Represents the user's identifier. It can be an IRI, email, or username.
-  * @param value the user's identifier.
+  * The UserIdentifierADM factory object, making sure that all necessary checks are performed and all inputs
+  * validated and escaped.
   */
-case class UserIdentifierADM(iri: Option[IRI] = None,
-                             email: Option[String] = None,
-                             username: Option[String] = None) {
+object UserIdentifierADM {
+    def apply(maybeIri: Option[String] = None,
+              maybeEmail: Option[String] = None,
+              maybeUsername: Option[String] = None)(implicit sf: StringFormatter): UserIdentifierADM = {
 
+        val parametersCount: Int = List(
+            maybeIri,
+            maybeEmail,
+            maybeUsername
+        ).flatten.size
 
-    val parametersCount: Int = List(
-        iri,
-        email,
-        username
-    ).flatten.size
+        // something needs to be set
+        if (parametersCount == 0) throw BadRequestException("Empty user identifier is not allowed.")
 
-    // something needs to be set
-    if (parametersCount == 0) throw BadRequestException("Empty user identifier is not allowed.")
+        if (parametersCount > 1) throw BadRequestException("Only one option allowed for user identifier.")
 
-    if (parametersCount > 1) throw BadRequestException("Only one option allowed for user identifier.")
+        new UserIdentifierADM(
+            maybeIri = sf.validateAndEscapeOptionalUserIri(maybeIri, throw BadRequestException(s"Invalid user IRI $maybeIri")),
+            maybeEmail = sf.validateAndEscapeOptionalEmail(maybeEmail, throw BadRequestException(s"Invalid email $maybeEmail")),
+            maybeUsername = sf.validateAndEscapeOptionalUsername(maybeUsername, throw BadRequestException(s"Invalid username $maybeUsername")))
+    }
+}
+
+/**
+  * Represents the user's identifier. It can be an IRI, email, or username.
+  * @param value the user's identifier.
```

Comment:
Could you update the parameters in the Scaladoc comment?

Reply:
done